pennylane/labs/resource_estimation/ops/identity.py

```diff
@@ -48,3 +62,19 @@ def resource_params(self) -> dict:
     @classmethod
     def resource_rep(cls, **kwargs) -> re.CompressedResourceOp:
         return re.CompressedResourceOp(cls, {})
+
+    @staticmethod
+    def adjoint_resource_decomp() -> Dict[re.CompressedResourceOp, int]:
+        return {}
+
+    @staticmethod
+    def controlled_resource_decomp(
+        num_ctrl_wires, num_ctrl_values, num_work_wires
+    ) -> Dict[re.CompressedResourceOp, int]:
+        if num_ctrl_values == 0:
+            return {re.ResourcePhaseShift.resource_rep(): 1}
+        raise re.ResourcesNotDefined
```

Comment:
Do we not need to check `if num_ctrl_wires == 1`? or do all multi controlled global phases decompose to the same phase shift operator?

Reply:
I modified the decomposition to be general now